packages/agent/tests/prototyping/clients/dwn-server-info-cache.spec.ts

```diff
@@ -0,0 +1,119 @@
+
+    const exampleInfo:ServerInfo = {
+      maxFileSize              : 100,
+      webSocketSupport         : true,
+      registrationRequirements : []
+    };
+
+    after(() => {
+      sinon.restore();
+    });
+
+    beforeEach(() => {
+      clock = sinon.useFakeTimers();
+      cache = new DwnServerInfoCacheMemory();
+    });
+
+    afterEach(async () => {
+      await cache.clear();
+      await cache.close();
+      clock.restore();
+    });
+
+    it('sets server info in cache', async () => {
+      const key1 = 'some-key1';
+      const key2 = 'some-key2';
+      await cache.set(key1, { ...exampleInfo });
+      await cache.set(key2, { ...exampleInfo, webSocketSupport: false }); // set to false
+
+      const result1 = await cache.get(key1);
+      expect(result1!.webSocketSupport).to.deep.equal(true);
+      expect(result1).to.deep.equal(exampleInfo);
+
+      const result2 = await cache.get(key2);
+      expect(result2!.webSocketSupport).to.deep.equal(false);
+    });
+
+    it('deletes from cache', async () => {
+      const key1 = 'some-key1';
+      const key2 = 'some-key2';
+      await cache.set(key1, { ...exampleInfo });
+      await cache.set(key2, { ...exampleInfo, webSocketSupport: false }); // set to false
+
+      const result1 = await cache.get(key1);
+      expect(result1!.webSocketSupport).to.deep.equal(true);
+      expect(result1).to.deep.equal(exampleInfo);
+
+      const result2 = await cache.get(key2);
+      expect(result2!.webSocketSupport).to.deep.equal(false);
+
+      // delete one of the keys
+      await cache.delete(key1);
+
+      // check results after delete
+      const resultAfterDelete = await cache.get(key1);
+      expect(resultAfterDelete).to.equal(undefined);
+
+      // key 2 still exists
+      const result2AfterDelete = await cache.get(key2);
+      expect(result2AfterDelete!.webSocketSupport).to.equal(false);
+    });
+
+    it('clears cache', async () => {
+      const key1 = 'some-key1';
+      const key2 = 'some-key2';
+      await cache.set(key1, { ...exampleInfo });
+      await cache.set(key2, { ...exampleInfo, webSocketSupport: false }); // set to false
+
+      const result1 = await cache.get(key1);
+      expect(result1!.webSocketSupport).to.deep.equal(true);
+      expect(result1).to.deep.equal(exampleInfo);
+
+      const result2 = await cache.get(key2);
+      expect(result2!.webSocketSupport).to.deep.equal(false);
+
+      // delete one of the keys
+      await cache.clear();
+
+      // check results after delete
+      const resultAfterDelete = await cache.get(key1);
+      expect(resultAfterDelete).to.equal(undefined);
+      const result2AfterDelete = await cache.get(key2);
+      expect(result2AfterDelete).to.equal(undefined);
+    });
+
+    it('returns undefined after ttl', async function () {
+      // skip this test in the browser, sinon fake timers don't seem to work here
+      // with a an await setTimeout in the test, it passes.
+      if (!isNode) {
```

Comment:
Sanity: this would be the first precedence that it doesn't work in browser right? Have you tried passing `number` instead of `string` to `tickAsync()`?

Reply:
It's really strange that it doesn't work. I just tried a number instead of string and get the same behavior because I didn't think to try that before. But still fails in the browsers.